### DISCUSSION TITLE
2 levels nested array field

### DISCUSSION THREAD

**Discussion by Skia69** [react-hook-form/react-hook-form#8266] [Q&A] [answered]
Hello, i'm sorry if this question has been answered before. I'm trying to build the form that i've shown in the screenshot below

So basically, i'll have an array of rounds and each round will have an array of scrims

I checked the useFieldArray docs but i couldn't really wrap my head around coming up with a 2 levels nested array field. Could someone be kind enough to show me how to do it?

Thanks in advance x

![Screenshot 2022-04-28 212444](https://user-images.githubusercontent.com/42115721/165835599-09d5fc35-8de2-4b65-9d63-3b803e7c701e.png)

**Comment by Moshyfawn** [comment, marked as answer]
Something along the lines of [this example](https://codesandbox.io/s/zealous-tess-jpwm9r?file=%2Fsrc%2FApp.tsx)

**Reply by Skia69** [reply]
Thank you for your response

Is there a chance you could extend your example and show me how do i add more `Field` and `Child` dynamically?

**Reply by Skia69** [reply]
@Moshyfawn You're godsend, thank you!

**Reply by Moshyfawn** [reply]
Here you go: I've updated [the previous example](https://codesandbox.io/s/zealous-tess-jpwm9r?file=%2Fsrc%2FApp.tsx) to include append / remove field array operations

**Reply by Skia69** [reply]
I'm sorry for bugging you, it's just that the nested useFieldArray thing is a bit hard for me to grasp

I have one more question - how do i replace `fields` and `children` with `rounds` and `scrim` ? 
I'm talking about the actual values that are logged in the console

I tried to tinker around with it but it kept on breaking 🤣

**Reply by Moshyfawn** [reply]
No worries! We are here to help you lol :P

The main thing you want to look at is the field's path _(and the types, if you're using TS)_. All the other stuff, like resigning the methods and such _(like `fields: children` or `append: appendChild`)_ is just my preference to be explicit about what the things do.

Here's a quick screen-cast. Obviously, you can omit / rename the methods and fields re-assignment. But to update the fields paths, here's how to do it:

https://user-images.githubusercontent.com/16290753/165973036-ec4d8eef-5af4-47a1-92f2-d7d72dfc8b84.mp4

P.S. I've updated [the example codesanbox](https://codesandbox.io/s/zealous-tess-jpwm9r?file=%2Fsrc%2FApp.tsx) to include `rounds` and `scrim` instead
